tests/generation/test_utils.py

```diff
@@ -1497,6 +1497,54 @@ def test_generate_with_head_masking(self):
                 attn_weights = out[attn_name] if attn_name == attention_names[0] else out[attn_name][-1]
                 self.assertEqual(sum([w.sum().item() for w in attn_weights]), 0.0)
 
+    # TODO (joao): this test is actually not slow :) However, it is not passing in some models (e.g. GPTNeoX) and the
+    # fix for some models is quite lengthy. Being slow means it doesn't block our push CI while we fix it.
+    @slow
+    def test_left_padding_compatibility(self):
+        # The check done in this test is fairly difficult -- depending on the model architecture, passing the right
+        # position index for the position embeddings can still result in a different output, due to numerical masking.
```

Comment:
just curious, why we use the term `numerical` here and below? Is `attention mask` not enough?

Reply:
I want to make it explicit that attention mask is not a binary operation, but rather a numerical operation (and thus constrained to the maximum negative value of the variable format of choice). Generation is quite sensible to masking, especially at lower resolution representations with limited dynamic range like `float16`.